### DISCUSSION TITLE
Mapping to transcriptome

### DISCUSSION THREAD

**Discussion by Jelinek-J** [alexdobin/STAR#1352] [Q&A] [answered]
Hi Alex,

I try to use STAR 2.7.9a for alignment to transcriptome (chapter 6 in doc), however I have a few question...
-  File `Log.final.out` contains statistics about unmapped reads for alignment to genome (multimapping, mismatches, etc.), however it does not contain any stats related to transcriptome alignment (unknown RNAME, outside exons, indels etc.). Are that statistics somewhere else or is it possible to get the stats somehow?
-  Would it be possible to add default `@CO` and `@PG` lines also into `*toTranscriptome*` files? In comparison to `@SQ` lines, it does not increase the file size too much and it would be usefull to log the command even in that file to does not care about corresponding `*sortedByCoord*` file.
-  Would be sometimes in future also included `--quantTranscriptomeBan SoftclipSingleend` (indels allowed) and  `--quantTranscriptomeBan IndelSingleend` (soft-clipping allowed) options and maybe also the other combinations?
-  Single-end alignments (see `--quantTranscriptomeBan`) are related for paired-end reads only?
-  And actually, how does e.g. a filter `--readStrand Forward` behave, is it applied when aligned to genome only, or when translated to transcripts (i.e. if a transcript is in the reverse strand, then some reads could be in `*sortedByCoord*` file and missing in `*toTranscriptome*` and on the contrary other reads will be in `*toTranscriptome*` and not in `*sortedByCoord*`)?

Thanks.

(Edited as part of the problem was solved in 2021-06-25 version - quite confusing that '2.7.9a' does not fully identifies the version.)

**Comment by alexdobin** [comment, marked as answer]
Hi @Jelinek-J 

these are good suggestions, thanks - will add them to my TODO list.

>Single-end alignments (see --quantTranscriptomeBan) are related for paired-end reads only?
```--quantTranscriptomeBan Singleend``` will prohibit single-end alignment for paired-end libraries.

```--readStrand Forward``` is not implemented at the moment.
Both "sense" and "antisense" alignment are output as transcriptomic alignments.

Cheers
Alex

**Reply by Jelinek-J** [reply]
Hi @alexdobin,
thank you for your answer.

And I realized one related point, so I attach it here: would it be also possible to add all relevant SAM attributes (`--outSAMattributes`) also to `*toTranscriptome*` files? Currently, there are only NH and HI attributes; rest of attributes are in `*sortedByCoord*` files only. I especially miss AS attribute (to update primary flag if an additional filtering is done) and MD attribute.

Thanks.
